src/wolfio.c

```diff
@@ -2714,4 +2714,385 @@ int wolfSSL_SetIO_LwIP(WOLFSSL* ssl, void* pcb,
+        word16 length)
+{
+    int ret;
+    ctx->sequence = 0;
+    ctx->flow_packets = 0;
+    ctx->flow_counter = 0;
+    /* First frame has 1 nibble for type, 3 nibbles for length followed by
+     * 6 bytes for data*/
+    ctx->frame.data[0] = (length >> 8) | (ISOTP_FRAME_TYPE_FIRST << 4);
+    ctx->frame.data[1] = length & 0xff;
+    XMEMCPY(&ctx->frame.data[2], buf, ISOTP_FIRST_FRAME_DATA_SIZE);
+    ctx->buf_ptr = buf + ISOTP_FIRST_FRAME_DATA_SIZE;
+    ctx->buf_length = length - ISOTP_FIRST_FRAME_DATA_SIZE;
+    ctx->frame.length = ISOTP_CAN_BUS_PAYLOAD_SIZE;
+    ret = ctx->send_fn(&ctx->frame, ctx->arg);
+    if (ret <= 0) {
+        WOLFSSL_MSG("ISO-TP error sending first frame");
+        return WOLFSSL_CBIO_ERR_GENERAL;
+    }
+    while(ctx->buf_length) {
+        /* The receiver can set how often to get a flow control packet. If it
+         * is time, then get the packet. Note that this will always happen
+         * after the first packet */
+        if (ctx->flow_counter == ctx->flow_packets) {
+            ret = isotp_receive_flow_control(ctx);
+        }
+        /* Frame delay <= 0x7f is in ms, 0xfX is X * 100 us */
+        if (ctx->frame_delay) {
+            if (ctx->frame_delay <= ISOTP_MAX_MS_FRAME_DELAY) {
+                ctx->delay_fn(ctx->frame_delay * 1000);
+            } else {
+                ctx->delay_fn((ctx->frame_delay & 0xf) * 100);
+            }
+        }
+        switch (ret) {
+            /* Clear to send */
+            case ISOTP_FLOW_CONTROL_CTS:
+                if (isotp_send_consecutive_frame(ctx) < 0) {
+                    WOLFSSL_MSG("ISO-TP error sending consecutive frame");
+                    return WOLFSSL_CBIO_ERR_GENERAL;
+                }
+                break;
+            /* Receiver says "WAIT", so we wait for another flow control
+             * packet, or abort if we have waited too long */
+            case ISOTP_FLOW_CONTROL_WAIT:
+                ctx->wait_counter += 1;
+                if (ctx->wait_counter > ISOTP_DEFAULT_WAIT_COUNT) {
+                    WOLFSSL_MSG("ISO-TP receiver told us to wait too many"
+                            " times");
+                    return WOLFSSL_CBIO_ERR_WANT_WRITE;
+                }
+                break;
+            /* Receiver is not ready to receive packet, so abort */
+            case ISOTP_FLOW_CONTROL_ABORT:
+                WOLFSSL_MSG("ISO-TP receiver aborted transmission");
+                return WOLFSSL_CBIO_ERR_WANT_WRITE;
+            default:
+                WOLFSSL_MSG("ISO-TP got unexpected flow control packet");
+                return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+    }
+    return 0;
+}
+
+int ISOTP_Send(WOLFSSL* ssl, char* buf, int sz, void* ctx)
+{
+    int ret;
+    struct isotp_wolfssl_ctx *isotp_ctx;
+    (void) ssl;
+
+    if (!ctx) {
+        WOLFSSL_MSG("ISO-TP requires wolfSSL_SetIO_ISOTP to be called first");
+        return WOLFSSL_CBIO_ERR_TIMEOUT;
```

Comment:
Is `WOLFSSL_CBIO_ERR_TIMEOUT` the right error code here? Maybe `WOLFSSL_CBIO_ERR_GENERAL`?